src/serial.rs

```diff
@@ -881,9 +881,7 @@ where
 {
     let isr = usart.isr.read();
 
-    Err(if isr.busy().bit_is_set() {
```

Comment:
Also, clarification for myself. This function would still return `nb::Error::WouldBlock` in case this bit is set / nothing is in the RDR register, because this case is handled by the `else` branch now.